### ISSUE TITLE
`mapStateToProps` could be confusing?

### ISSUE THREAD

**Issue by winkler1** [gaearon/redux#431]
(moved [from here](https://github.com/jackielii/simplest-redux-example/pull/3))

The first arg to `connect` in `react-redux` is called `mapStateToProps`.

```
// Map Redux state to component props
function mapStateToProps(state)  {
  return {
    value: state.count
  };
}

// Connected Component:
let App = connect(
  mapStateToProps
)(Counter);
```

This the the edge where `Redux` meets `React`. Both can have state. The `connect` function is wrapping a React component, so people are likely thinking in React terms. `"state"` is ambiguous.

 To be painfully explicit, the method could include the nouns: `mapStoreStateToComponentProps`.  Which is too verbose. `mapStoreToComponent` makes more sense IMO.

In general, seems best to avoid ambiguous terms -- people get confused. Maybe this is a mindset change to have Redux own state, which I absolutely love. But am just imagining a lot of discussion with confused coworkers.

**Comment by gaearon** [comment]
It's better to discuss this in `react-redux` than here, but I won't ask you to move the issue again :-P
This is my argument for keeping the current naming.

It's given `state`, not `store`. Store is a specific thing in Redux, and it's *not* what you get in this function. React people *will* need to learn about Store's state sooner or later. Store has `getState()` method. Reducers receive `state` as argument. It's all over the documentation. Calling it something else in React bindings IMO is even more misleading.

In a function `mapStoreToProps` I'd expect to be able to bind action creators, because `dispatch` is available on the Store. It's Store's method. However there's a separate `mapDispatchToProps` for some reason..

In short I think it's easier to get people to learn about Redux's approach to state first, and then keep consistent naming. [State is introduced very early](http://gaearon.github.io/redux/docs/introduction/ThreePrinciples.html), and it's called "state" in every other Redux doc.

**Comment by gaearon** [comment]
I'm not inclined to change `state` to `store` because we call it `state` everywhere else. Better to confuse once, than to confuse many times.

**Comment by zealoushacker** [comment]
Would it be good to have an `ownState` param, much like there's an `ownProps` param to remove any confusion?

**Comment by markerikson** [comment]
Uh... what would that even mean?  I'm not following what you're suggesting.
